lib/number.js

```diff
@@ -21,6 +22,11 @@ module.exports = function (options, input, {
       else return 0;
     } else {
       if (!allowDecimals) return Math.floor(input);
+
+      if (input === +input && input !== (input|0) && withPrecision) {
```

Comment:
If we check `withPrecision` first, we won't have to check any other logic if it is `false`.

```js
if (withPrecision && input === +input && input !== (input|0))
```

If you're attempting to convert input to an integer, it seems[ `Math.floor` is actually faster](https://jsperf.com/or-vs-floor/2).

Overall, awesome feature!